### PR TITLE
Update dependencies (mio 0.6, slab 0.3)

### PR DESCRIPTION
**Pull request by DenisKolodin**
This PR updates dependencies of this crate.
I've done it to prepare the crate for some experiments with #56 

All changes keeps backward compatibility, but some implementations changed. I've changed `accept` methods, because `mio` has changes with API and most of structs used by this crate had deprecated.
Let's consider this PR as **keep crate alive** changes and as first step towards the **new mio**'s API.

### CODE REVIEW

**Comment by zonyitoo** [comment]
Thanks

**Comment by zonyitoo** [comment]
Hmm, I saw `Reflect` has been deprecated in the latest rustc. And `coroutine_size` test is failed, don't know why.

**Comment by DenisKolodin** [comment]
Thanks for merging!
I saw this fail before changes, but haven't try to fix it, because I thought it's normal behavior.
Now I see we have to fix it too. I'll open an Issue to track this problem.

**Comment by lhecker** [comment]
@DenisKolodin The real socket tests (those in the "tests" folder) fail with EAGAIN now though - I believe your PR might not have been a perfect update. 😅

**Comment by DenisKolodin** [comment]
@lhecker Wow! Why all tests hadn't ran? I thought `cargo test` runs all tests we have :cold_sweat:
Any ideas how to fix? Actually `mio` has large API changes, It seems we should move from deprecated to a new or we have to repair it with deprecated?

**Comment by zonyitoo** [comment]
If mio deprecated all its IO wrappers to `::deprecated` mod, did they provide some other new wrappers for this?

I saw `Error { repr: Os { code: 11, message: "Resource temporarily unavailable" }` in the panic message. WHY?

File: src/net/tcp.rs [hunk 1]
```diff
@@ -42,33 +41,14 @@ impl TcpListener {
...
-            sync_guard.disarm();
-
-            match *self.read_timeout.lock() {
-                None => self.ready_states.wait(ReadyType::Readable),
-                Some(t) => {
-                    if self.ready_states.wait_timeout(ReadyType::Readable, t) {
-                        return Err(make_timeout());
-                    }
-                }
+            Err(err) => {
```
Comment (zonyitoo) [inline src/net/tcp.rs]: You have to handle `err.kind() == ErrorKind::WouldBlock`!!!

File: src/net/unix.rs [hunk 2]
```diff
@@ -87,33 +86,14 @@ impl UnixListener {
...
-            sync_guard.disarm();
-
-            match *self.read_timeout.lock() {
-                None => self.ready_states.wait(ReadyType::Readable),
-                Some(t) => {
-                    if self.ready_states.wait_timeout(ReadyType::Readable, t) {
-                        return Err(make_timeout());
-                    }
-                }
+            Err(err) => {
```
Comment (zonyitoo) [inline src/net/unix.rs]: The same as about, you have to handle `WouldBlock`

**Comment by lhecker** [comment]
@DenisKolodin I believe `cargo test` only runs system/end-to-end tests when all unit tests pass. Could you or @zonyitoo add `--no-fail-fast` to the `.travis.yml` file in the next commit?

**Comment by zonyitoo** [comment]
I would revert this PR. Please @DenisKolodin handles `ErrorKind::WouldBlock` properly.

**Comment by DenisKolodin** [comment]
Ok, I'll make it right.

**Comment by lhecker** [comment]
@zonyitoo @DenisKolodin I fixed it real quick just now. 🙂

**Comment by zonyitoo** [comment]
Awesome!

**Comment by DenisKolodin** [comment]
@lhecker Thanks a lot! :heart_eyes: But what about `--no-fail-fast` for travis and appveyor? :innocent: I can add it if you want.

**Comment by DenisKolodin** [comment]
PR #64 adds these flags

**Pull request merged**